### PR TITLE
feat: add WHATWG-aligned HTML AST parser

### PR DESCRIPTION
**Pull request by aryanraj45**
**Summary**

This PR implements a high-performance, WHATWG-aligned HTML AST tree builder and integrates it into `HtmlParser.js`. It models the AST structures closely after `swc_html_parser` and includes a simplified tree construction algorithm that handles essential rules (e.g., text merging, implicit closing of `<p>`, self-closing pairs like `<li>`, and SVG/MathML namespaces). 

The AST is attached to `module.buildInfo.htmlAst` to establish the core compiler foundation for future HTML minification and transpilation plugins. It runs securely alongside the existing token extractor, ensuring 100% backward compatibility. (Complex edge-case algorithms like Foster Parenting and the Adoption Agency Algorithm are marked with TODOs for follow-up PRs).

**What kind of change does this PR introduce?**

feat

**Did you add tests for your changes?**

Yes. Added 100% test coverage for the AST builder in `test/buildHtmlAst.unittest.js`. Existing HTML config cases were also fully verified to pass without regression.

**Does this PR introduce a breaking change?**

No.

**If relevant, what needs to be documented once your changes are merged or what have you already documented?**

n/a

**Use of AI**

Partial

### CODE REVIEW

**Comment by aryanraj45** [comment]
hey @alexander-akait address review the co-pilot comments and implemented AST praser (nodes, SVG, auto-close), with complex edge cases (Adoption Agency, Foster Parenting) marked with TODOs for next PR.Please have a look thanks:))

File: lib/html/HtmlParser.js [hunk 1]
```diff
@@ -1480,6 +1481,9 @@ class HtmlParser extends Parser {
 		const buildMeta = /** @type {BuildMeta} */ (state.module.buildMeta);
 		buildMeta.exportsType = "default";
 
+		// Attach the WHATWG-aligned HTML AST to the build info for future minification and transpilation
+		buildInfo.htmlAst = buildHtmlAst(source);
```
Comment (alexander-akait) [inline lib/html/HtmlParser.js]: Do not store it here right now, we don't need extra memory consuming we don't use it right now

Comment (aryanraj45) [reply lib/html/HtmlParser.js]: @alexander-akait Done can u please take a look thanks :))

**Comment by alexander-akait** [comment]
Please fix lint

**Comment by aryanraj45** [comment]
@alexander-akait done can u please have a look lint is fixed and re-run the ci .sorry previous comment was deleted by mistake 
image
